clients/src/main/java/org/apache/kafka/common/record/ControlRecordType.java

```diff
@@ -43,9 +43,13 @@
 public enum ControlRecordType {
     ABORT((short) 0),
     COMMIT((short) 1),
+
     // Raft quorum related control messages.
-    QUORUM_REASSIGNMENT((short) 2),
+    // TODO - ngoel
+    // Burn enum val 2 to prevent
+    // accidental corruption. (Consult with team)
```

Comment:
The change proposed in this PR  already a backward incompatible change. I think we agreed to make `LEADER_CHANGE(2)` and `SNAPSHOT_HEADER(3)`.

Reply:
Makes sense. Made the change in the next revision